ads/google/a4a/utils.js

```diff
@@ -357,92 +356,71 @@ export function additionalDimensions(win, viewportSize) {
 /**
  * Extracts configuration used to build amp-analytics element for active view.
  *
+ * @param {!../../../extensions/amp-a4a/0.1/amp-a4a.AmpA4A} a4a
  * @param {!../../../src/service/xhr-impl.FetchResponseHeaders} responseHeaders
  *   XHR service FetchResponseHeaders object containing the response
  *   headers.
- * @param {!../../../src/service/extensions-impl.Extensions} extensions
- * @return {?AmpAnalyticsConfigDef} config or null if invalid/missing.
+ * @return {?JSONType} config or null if invalid/missing.
  */
-export function extractAmpAnalyticsConfig(responseHeaders, extensions) {
+export function extractAmpAnalyticsConfig(a4a, responseHeaders) {
   if (responseHeaders.has(AMP_ANALYTICS_HEADER)) {
```

Comment:
Rather than nesting in a large if, how about:

if (!responseHeaders.has(AMP_ANALYTICS_HEADER)) { return null; }
// contents of previous if